client/components/addAttendance/addAttendance.directive.js

```diff
@@ -25,6 +31,52 @@ angular.module('observatory3App')
     $scope.formTypeEmptyError = !$scope.attend.type;
   };
 
+  function getAttendanceStatus(data) {
+    var date = data.date,
+    mode = data.mode;
+    if (mode === 'day') {
+      var dayToCheck = new Date(date).toDateString();
+      for(var i = 0; i< $scope.user.attendance.length; i++){
+        var currentDay = new Date($scope.user.attendance[i].date).toDateString();
+        var isVerified = $scope.user.attendance[i].verified;
+        if(dayToCheck ===currentDay){
+          if(isVerified){
+            return 'attended';
+          } else {
+            return 'unverified';
+          }
+        }
+      }
+    }
+    return '';
+  }
+
+  function fixOffByOneDay(daystring){
```

Comment:
The value of user.attendance.date is a string with the form of yyyy-mm-ddT00:00:00Z and when you pass it into new Date(...), it returns a Date object of the day before so it messed up the comparison. Changed it to T04:00:00Z fixed this problem.